nilearn/_utils/niimg_conversions.py

```diff
@@ -161,176 +276,44 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
         A single image.
     """
 
-    # get properties from first image
+    target_fov = 'first' if auto_resample else None
+
+    # First niimg is extracted to get information and for new_img_like
+    first_niimg = None
+
+    iterator, literator = itertools.tee(iter(niimgs))
     try:
-        first_niimg = check_niimg(next(iter(niimgs)), atleast_4d=True)
+        first_niimg = check_niimg(next(literator))
     except StopIteration:
         raise TypeError('Cannot concatenate empty objects')
```

Comment:
I'm not sure. We do not displayed it because it would be either "[]" or nothing (in the case of a functional iterator). So I don't think that it is very useful.